api/policies/externalAuth.js

```diff
@@ -0,0 +1,27 @@
+const { Op } = require('sequelize');
+const { Site } = require('../models');
+
+module.exports = function externalAuth(req, res, next) {
+  if (req.query && req.query.site_id) {
```

Comment:
This request is coming from Netlify CMS so in the current setup it would not know 1. The `site_id`, is your intention that Federalist adds this query parameter when it is responsible for providing the backend configuration during the build? This implies that users would only be able to use Netlify CMS on a Federalist deployed site and not when developing locally.

I would think that just using the repo name would be enough to identify a site without exposing/requiring the use of the id which is more of an implementation detail.

2. Is there a strong reason for doing this check BEFORE authenticating with Github? I don't think is necessarily a security risk with the private buckets, but it does expose some internal information about the deployed site before Federalist knows whether or not the user should have access to it. My thought would be to handle this after we know that the user has authenticated with Github AND has access to Federalist -> the `passport` "external callback". This is also where the Netlify CMS UI is expecting to receive an authentication success or failure so might result in a better UX without more work.

Reply:
the site_id is delivered automatically as a parameter in the authentication request  .... how would you advise getting these repo/site_id details from the external callback? i looked but did not see this in the callback

Reply:
1. I'm a bit confused, what am I missing? How does Netlify CMS know the Federalist site_id?
2. i don't totally have the answer on the callback, we could potentially embed it in the "state" parameter that we send as part of the oauth request, which is then returned as proof that the callback was initiated by us...

Reply:
netlify is sending it automatically for all auth requests.  it takes a window.location and sends it in site_id ...

hmm ... the state param ... that's not a bad idea ... i have to look into that .... i think we would still need the site_id to get the repository ... we could do a pre and post auth check ... if we want to get this working on localhost we could add the owner and repository to the extrnal_auth path?

Reply:
are you sure site_id is what you expect? this is what it is locally:
```https://federalistapp-staging.18f.gov/external/auth/github?provider=github&site_id=cms.netlify.com&scope=repo```

Reply:
oooooh, so site_id is the domain?

Reply:
I understand this now. But i'm wondering if this is potentially subject to abuse. We are currently only checking that the user is in an "approved" Github org, so even if we move the site_id check to after, a malicious user within an approved org for a site on a shared bucket, could pass a different site_id that corresponds to one with a dedicated bucket allowing access to Netlify CMS.

Perhaps we can do this along with verifying that the user has access to the Federalist site? Right now, any user that is in an approved Federalist org can access Federalist, but to access a Federalist site, do have to have Github permissions for that repo?